pkg/probe/http/http_test.go

```diff
@@ -87,13 +90,156 @@ func TestHTTPProbeProxy(t *testing.T) {
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404},
+			expectHTTPContent: "",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{500, 404},
+			expectHTTPContent: "",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{200},
+			expectHTTPContent: "",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{400, 200},
+			expectHTTPContent: "",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404, 500},
+			expectHTTPContent: "not*",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404, 500},
+			expectHTTPContent: "found*",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{200},
+			expectHTTPContent: "not *",
+			health:            probe.Failure,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{404, 500},
+			expectHTTPContent: "not *",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "not *",
+			health:            probe.Success,
+		},
+		{
+			handler:           handleReq(http.StatusNotFound, "not found"),
+			expectHTTPCodes:   []int{},
+			expectHTTPContent: "found *",
+			health:            probe.Failure,
+		},
+	}
+
+	for i, test := range testCases {
+		t.Run(fmt.Sprintf("case-%2d", i), func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				test.handler(w, r)
+			}))
+			defer server.Close()
+			u, err := url.Parse(server.URL)
+			if err != nil {
+				t.Errorf("case %d: unexpected error: %v", i, err)
+			}
+			_, port, err := net.SplitHostPort(u.Host)
+			if err != nil {
+				t.Errorf("case %d: unexpected error: %v", i, err)
+			}
+			_, err = strconv.Atoi(port)
+			if err != nil {
+				t.Errorf("case %d: unexpected error: %v", i, err)
+			}
+			health, output, err := prober.Probe(u, test.reqHeaders, test.expectHTTPCodes, test.expectHTTPContent, 1*time.Second)
+			//health, output, err := prober.Probe(u, test.reqHeaders, test.expectHTTPCodes, test.expectHTTPContent, 1*time.Second)
```

Comment:
I'd suggest to remove commented out code.

Reply:
done